### DISCUSSION TITLE
Test Driven Development

### DISCUSSION THREAD

**Discussion by gunar** [dubzzz/fast-check#3382] [Q&A] [answered]
Hey guys,

I'm wondering how y'all are integrating Property Based Testing (PBT) and Test Driven Development (TDD). 

- When I'm not using fast-check (i.e. when I write standardized tests) it's easier to write simpler, smaller tests which get the TDD cycle really going.
- With PBT, I have a _hunch_ that I tend to not write those very small and simple tests and immediately write bigger tests that cover a bigger chunk of the desired behavior, which is antithetical to TDD.

What's your experience?

**Comment by dubzzz** [comment, marked as answer]
Hey,

I don't have a direct answer to your question, but you may have look at some PBT I wrote to get an idea of a way to use it: https://github.com/dubzzz/advent-of-pbt-2021
The associated articles are available at: https://dev.to/dubzzz/advent-of-pbt-2021-13ee
